File: app/src/main/java/org/techtown/samplerecorder/Queue.java
package org.techtown.samplerecorder;

import java.io.File;
import java.nio.Buffer;
import java.nio.ByteBuffer;
import java.nio.ShortBuffer;
import java.util.LinkedList;

public class Queue {

    java.util.Queue<ShortBuffer> queue;

    public Queue() {
        myLog.d("constructor activate");

        queue = null;
        queue = new LinkedList<ShortBuffer>();
    }

    public void enqueue(ShortBuffer shortBuffer) {
        myLog.d("method activate");

        queue.add(shortBuffer);
    }

    public ShortBuffer dequeue() {
        myLog.d("method activate");

        return queue.peek();
    }

}

File: app/src/main/java/org/techtown/samplerecorder/Audio/Queue.kt
package org.techtown.samplerecorder.Audio

import java.util.*
import java.util.Queue

class Queue {
    var queue: Queue<ByteArray>? = null
    var store: Queue<ByteArray>

    fun enqueue(data: ByteArray) {
        queue!!.offer(data)
        store.offer(data)
    }

    @Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
    fun dequeue(): ByteArray {
        return queue!!.poll()
    }

    fun copy() {
        queue!!.addAll(store)
    }

    val isEmpty: Boolean
        get() = queue!!.isEmpty()

    init {
        queue = LinkedList()
        store = LinkedList()
    }

    companion object {
        private const val TAG = "Queue"
    }
}